pkg/controller/util.go

```diff
@@ -594,3 +601,14 @@ func IsPopulated(pvc *v1.PersistentVolumeClaim, c client.Client) (bool, error) {
 		return dv, err
 	})
 }
+
+// SetPodPvcAnnotations applies PVC annotations on the pod
+func SetPodPvcAnnotations(pod *v1.Pod, pvc *v1.PersistentVolumeClaim) {
+	allowedAnnotations := []string{AnnPodNetwork, AnnPodSidecarInjection}
+	for _, ann := range allowedAnnotations {
+		if val, ok := pvc.Annotations[ann]; ok {
+			klog.V(1).Info("Applying PVC annotation on the pod", ann, val)
+			pod.ObjectMeta.Annotations[ann] = val
```

Comment:
`pod.ObjectMeta.Annotations` may be `nil`.  Probably not in practice but definitely in theory

Reply:
So i'm adding an `if`